### DISCUSSION TITLE
Extract and display a cover image for each article

### DISCUSSION THREAD

**Discussion by luben** [osmoscraft/osmosfeed#15] [Ideas]
I would appreciate if there is an option, e.g. set in `osmosfeed.yaml` to display unmodified `content:encoded` if available (from the RSS feed itself). I am not sure if it's related but I will also appreciate showing the images, e.g. XKCD strips from https://xkcd.com/rss.xml, inline.

**Comment by chuanqisun** [comment]
Thanks for the feedback @luben. I've been considering extracting a "cover image" from the article content. Would that address your core issue? Or there is something else in the unmodified encoded content you'd need?

**Reply by onnyyonn** [reply]
I have no idea how difficult it is, but if a relevant media file (image / audio / video) can be extracted and embedded, that would be great. For example, a YouTube link will embed a video, a podcast link will embed an audio etc. Also, it would be cool to have watching time or listening time for them instead of reading time.

**Reply by chuanqisun** [reply]
It is not trivial but I think this would be a perfect use case for a plugin! This also gives me some thought about how to design the upcoming plugin SDK. It is three sided problem:
- How to extend the backend, so the crawler can get additional data, without conflict with other plugins
- How to extend the UI, so the additional data can be displayed, without conflict with other plugins
- How to let the plugin developer expose some settings to the end user

I think WordPress theme/plugin ecosystem is a very successful example. I might borrow some ideas from them.

**Comment by luben** [comment]
Yes, that's one option. Also I noticed my first request didn't make a lot of sense (opposite of the title), so I edited it.

**Comment by 2m** [comment]
I would also love such feature. Browsing through my oldest RSS feeds, I see that some of the articles are no longer accessible. Having osmofeed store the content of the article would be perfect for archiving purposes.

**Comment by chuanqisun** [comment]
@2m , thank you for providing the motivation. I have converted your ask to #14.
